### DISCUSSION TITLE
Text exctraction

### DISCUSSION THREAD

**Discussion by erkin98** [jsvine/pdfplumber#377] [Q&A] [answered]
how can i extract text form particular area without surrounded lines?

**Comment by samkit-jain** [comment]
Hi @erkin98, could you please provide the PDF and provide more details on where that area is in the PDF?

**Comment by erkin98** [comment]
[https://drive.google.com/file/d/18smAX6VTvqbfyEQ6e20Tg6iOl3mDBMZF/view?usp=sharing](url)
in this pdf i want to crop areas top left as a table?is it possible or i just parse as text?

**Comment by samkit-jain** [comment, marked as answer]
Thanks for sharing the PDF @erkin98 Since, that top left text is not wrapped in a `rect` object, there is no straightforward way to extract it. One alternate workaround would be to get the coordinates of those horizontal lines at the top and bottom of the text, crop the page and then extract text from it.

You can do so by running the following code:
```python
import pdfplumber

pdf = pdfplumber.open("file.pdf")
page = pdf.pages[0]
top_line = page.horizontal_edges[2]  # The top line is actually the 3rd horizontal edge.
bottom_line = page.horizontal_edges[4]  # The bottom line is actually the 5th horizontal edge.

page = page.crop(
    (top_line["x0"], top_line["top"], top_line["x1"], bottom_line["bottom"])  # Create a rect object using the coordinates of the 2 horizontal edges.
)

print(page.extract_text())
```

The cropped page looks like
![image](https://user-images.githubusercontent.com/15127115/111463586-bf7c0000-8745-11eb-9b30-1de17a27648b.png)
and the output is
```
BAYBURT GRUP İNŞAAT NAKLİYAT MAD.İT.İHR.SAN.VE TİC.A.Ş.
IŞIK SOKAK  No:20 
06570 Çankaya/ Ankara 
Tel: 3122290808 Fax: 3122290808 
Web Sitesi: .
E-Posta: muhasebe@bayburtgroup.com
Vergi Dairesi: MALTEPE VD. 
TICARETSICILNO: 309029
MERSISNO: 0151043382200021
VKN: 1510433822
```

**Reply by samkit-jain** [reply]
@erkin98 I have selected the message as the answer based on the rocket emoji reaction from you.
